### DISCUSSION TITLE
"Code of conduct" not always showing with `CODE_OF_CONDUCT.md`

### DISCUSSION THREAD

**Discussion by julien-deramond** [community/community#52365] [Repositories]
### Select Topic Area

Bug

### Body

`CODE_OF_CONDUCT.md` is one of the right file names understood by GitHub to store codes of conduct in repositories as mentioned in https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-a-code-of-conduct-to-your-project#adding-a-code-of-conduct-using-a-template.

If I create this `CODE_OF_CONDUCT.md` from GitHub's interface, I can see that it is indeed understood since the interface suggests to me to start from a template:
![Screenshot 2023-04-09 at 12 00 13](https://user-images.githubusercontent.com/17381666/230766476-a0967bd2-6b0f-44c0-b989-2159984161e5.png)

However, when I don't use the template by clicking on "Choose a code of conduct template" but instead write my own content, when the file is committed, I can't see the following in the corresponding repository:

![Screenshot 2023-04-09 at 12 01 56](https://user-images.githubusercontent.com/17381666/230766538-49f8bfdc-4db8-44f5-9195-ad402ecf0b51.png)

Here are some examples of when this display is not available in the repository even if the projects have a `CODE_OF_CONDUCT.md` file at the root level:
- https://github.com/twbs/bootstrap
- https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap

On your side, in https://github.com/community/community, we can see this information because you probably generated it from the template.

In your documentation, it is written that "our code of conduct will be available either way, but 'Code of conduct' will only be marked as complete in your repository's community profile if you use a template.".

I would find it more helpful if it was always visible whatever the way used by the projects but maybe there's a technical limitation that I don't know that forces you to have this rule.

However, I can observe some weird behaviors. It's like what's important is rather the first time you create the code of conduct.
* If the file is named `CODE_OF_CONDUCT` (without the extension) and its content is created manually, I can see the information in the repository information area.
* If one repository creates the code of conduct from the template, deletes it, and recreates a `CODE_OF_CONDUCT.md` file from scratch with custom content. It works.
* If one repository creates the code of conduct from scratch as a `CODE_OF_CONDUCT.md`, deletes it, and uses the template. It doesn't work.

As a summary, depending on the context of the creation of this `CODE_OF_CONDUCT.md`, the "Code of conduct" information might not be displayed on the repository GitHub homepage. And it seems to be linked to the way it is created the first time.

**Comment by julien-deramond** [comment]
Nevermind, this random rendering comes from "Refined GitHub" browser extension. My apologies :)

**Comment by gpaulsen** [comment]
I'm also trying to create a `CODE_OF_CONDUCT.md` in my personal repos in Github Enterprise, but am not seeing the "Choose code of conduct from a template" button, while following https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-a-code-of-conduct-to-your-project.

I thought perhaps it was because I needed to first select "Template repository" from the Settings (see: https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-template-repository) but this did not help anything either.

:(

**Comment by Luedvi** [comment]
Something similar happened to me. It seems that if you create the CODE_OF_CONDUCT file from the template, the checkmark in the Insights>Community Standards is marked as complete, but if you modify the file after that, it turns back to "uncheck".
